include/packet.h

```diff
@@ -0,0 +1,14 @@
+#include <string>
+#include <vector>
+
+class Packet {
+    public:
+        struct ServerPacket {
+            int id;
+            std::vector<int> coor;
+            std::string message;
+        };
+
+        void serialize_server_packet(ServerPacket* pkt, char* buf);
+        void deserialize_server_packet(ServerPacket* pkt, char* buf);
```

Comment:
Could these be static? i.e., if we call using `somePacket.serialize(actualPacket, buf)`, how is the calling packet `somePacket` useful if this is not a static func?

Reply:
forgot static functions were a thing tbh